src/nexus-prisma/NexusPrismaBuilder.ts

```diff
@@ -112,17 +95,80 @@ export class NexusPrismaBuilder {
               mapping,
             }
           })
-
-          return this.buildSchemaForCRUD(t, 'Mutation', mutationFields)
+        } else {
+          throw new Error(
+            `t.crud can only be used on a 'Query' & 'Mutation' objectType. Please use 't.model' instead`,
+          )
         }
+
+        type FieldPublisher = (opts?: FieldPublisherConfig) => CRUDMethods // Fluent API
+        type CRUDMethods = Record<string, FieldPublisher>
+
+        return mappedFields.reduce<CRUDMethods>((crud, mappedField) => {
+          const prismaModelName = mappedField.mapping.model
+
+          mappedField.fields.forEach(field => {
```

Comment:
I expect that more refactoring will come down on this forEach block and all its contents as I get a bette handle on the systems/layers at play.

I suspect it would help a lot if we could introduce a few maps instead of one giant forEach. Worse for performance (if eager) but easier to reason about––but not sure, maybe teasing apart a few discrete stages here will prove illusive.

Reply:
The reason why it's a forEach is because we're calling `t.field`, which we cannot do in something else.
We could, however, have a map that'd precompute all the nexus configs, then have a `forEach` that'd just call the `t.field` and pass the config straight

Reply:
> We could, however, have a map that'd precompute all the nexus configs, then have a forEach that'd just call the t.field and pass the config straight

Yeah! Idea there I guess being we have one/a few pure data transformations then the final side-effect